bot/common/threads/onboarding.py

```diff
@@ -395,16 +388,24 @@ class PromptUserWalletMessageSignatureStep(BaseStep):
 
     name = StepKeys.PROMPT_USER_WALLET_ADDRESS.value
 
-    def __init__(self):
+    def __init__(self, cache, guild_id):
         super().__init__()
         self.prompt_for_verification = PromptForAccountVerification(
             self.get_account_verification_prompts
         )
+        self.cache = cache
+        self.guild_id = guild_id
 
     async def send(self, message, user_id):
-        return await self.prompt_for_verification.prompt(message), None
+        return await self.prompt_for_verification.prompt(user_id, message), None
 
-    def get_account_verification_prompts(self):
+    async def save(self, message, guild_id, user_id):
+        await self.verify_message(message)
+        await self.save_authenticated_account(user_id, guild_id)
+
+    async def get_account_verification_prompts(self, user_id):
+        # workaround to async-ify this method
+        await asyncio.sleep(0.001)
```

Comment:
ugly workaround. this is async, since the corresponding method in the twitter flow needs to be async, since the requested tweet w/nonce is stored in the cache. the reason they _both_ need to be async is in an attempt to DRY, the method signatures are aligned to fit the PromptForAccountVerification class, which actually isn't very useful anymore. the verification flows are sufficiently different at this point, that trying to align them just introduces needless complexity.

Reply:
Ah this makes sense, it may be worth having a tech debt ticket to untie the verification flows